sympy/parsing/sympy_parser.py

```diff
@@ -19,12 +19,15 @@
 from sympy.utilities.misc import func_name
 from sympy.functions.elementary.miscellaneous import Max, Min
 
+if TYPE_CHECKING:
+    from typing_extensions import TypeAlias
+
 
 null = ''
 
-TOKEN = tuple[int, str]
-DICT = dict[str, Any]
-TRANS = Callable[[list[TOKEN], DICT, DICT], list[TOKEN]]
+TOKEN: TypeAlias = tuple[int, str]
+DICT: TypeAlias = dict[str, Any]
+TRANS: TypeAlias = Callable[[list[TOKEN], DICT, DICT], list[TOKEN]]
```

Comment:
This change makes Sphinx want to see TRANS in the docs apparently. I'm not sure what the advantage is of having TypeAlias here rather than a plain assignment. Obviously in future we will want to use the type statement but minimum Python version is 3.9 right now.

Reply:
> I don't know either why this trips up Sphinx.

I don't know either.

> Maybe it's best to omit until 3.10 is the minimum supported Python version,

Why is 3.10 significant? I'm waiting for 3.12 and the type statement.

Reply:
> > Maybe it's best to omit until 3.10 is the minimum supported Python version,
> 
> Why is 3.10 significant? I'm waiting for 3.12 and the type statement.

3.10 is when `TypeAlias` was added to `typing`. (Which precedes 3.12's type keyword)
`typing_extensions.TypeAlias` being a backport.

I'm saying unless adding the `TypeAlias` type annotation explicitly solves an issue, it might be better that I don't care about it for this repo until it can be imported from the stdlib. Even if using it is "technically correct".

Reply:
> I'm saying unless adding the `TypeAlias` type annotation explicitly solves an issue

Are there any situations where it can solve an issue?

Reply:
It can fix false positives with https://mypy.readthedocs.io/en/stable/error_code_list.html#check-validity-of-types-valid-type

I don't remember the details off the top of my head, but in the last year I remember solving a mypy `valid-type` issue when trying to subclass some aliases in setuptools.

It happens, but very rare in my experience (I can't recall another case right now)